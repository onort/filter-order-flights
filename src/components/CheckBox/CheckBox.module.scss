/* stylelint-disable declaration-colon-newline-after */
@import "../../styles/main.scss";

.container {
  position: relative;
  display: block;
  padding-left:
    calc(
      #{map-get($font-sizes, normal)} + #{map-get($spacers, x-small)}
    );
  margin-bottom: map-get($spacers, small);
  color: $grey1;
  cursor: pointer;
  font-family: $font-family-primary;
  font-size: map-get($font-sizes, small);
  font-weight: 400;
}

.checkbox {
  position: absolute;
  width: 0;
  height: 0;
  cursor: pointer;
  opacity: 0;
}

.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  width: map-get($font-sizes, normal);
  height: map-get($font-sizes, normal);
  border: 1px solid $grey4;
  margin-right: map-get($spacers, x-small);
  background-color: $white;
  border-radius: px-to-rem(2px);

  &::after {
    position: absolute;
    display: none;
    content: "";
  }

  &.checked {
    background-color: $primary;
    border-color: $primary;

    &::after {
      top: px-to-rem(2px);
      left: px-to-rem(5px);
      display: block;
      width: px-to-rem(4px);
      height: px-to-rem(8px);
      border: solid $white;
      border-width: 0 1px 1px 0;
      transform: rotate(45deg);
    }
  }
}
