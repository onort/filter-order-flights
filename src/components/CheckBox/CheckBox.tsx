import React from "react"
import cx from "classnames"

import styles from "./CheckBox.module.scss"

interface Props {
  className?: string
  checked?: boolean
  label: string
  onChange: (e: React.InputHTMLAttributes<HTMLElement>) => void
}

const CheckBox: React.SFC<Props> = props => {
  const { checked, label, onChange } = props
  const classNames = cx(styles.container, props.className)
  const checkmarkClasses = cx(styles.checkmark, { [styles.checked]: checked })
  return (
    <label className={classNames}>
      {label}
      <input
        className={styles.checkbox}
        type="checkbox"
        checked={checked}
        onChange={onChange}
      />
      <span className={checkmarkClasses} />
    </label>
  )
}

CheckBox.defaultProps = {
  checked: false
}

export default CheckBox
