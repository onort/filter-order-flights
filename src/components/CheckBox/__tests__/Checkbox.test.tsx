import React from "react"
import { shallow, mount } from "enzyme"

import CheckBox from "../"

const mockChange = jest.fn()

describe("<CheckBox />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<CheckBox label="Test" onChange={mockChange} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have label rendered", () => {
    const testLabel = "Test"
    const wrapper = shallow(<CheckBox label="Test" onChange={mockChange} />)
    expect(wrapper.find("label").text()).toBe(testLabel)
  })

  it("should have checked class when checked prop has been passed", () => {
    const wrapper = shallow(
      <CheckBox label="Test" onChange={mockChange} checked={true} />
    )
    expect(wrapper.find(".checkmark").hasClass("checked")).toBe(true)
  })

  it("should fire onChange when input is clicked", () => {
    const handleChange = jest.fn()
    const wrapper = mount(<CheckBox label="Test" onChange={handleChange} />)
    wrapper.find("input").simulate("change", { target: { checked: true } })
    expect(handleChange).toBeCalledTimes(1)
    wrapper.unmount()
  })

  it("should be able to have a custom className", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(
      <CheckBox className={testClassName} label="Test" onChange={mockChange} />
    )
    expect(wrapper.find("label").hasClass(testClassName)).toBe(true)
  })
})
