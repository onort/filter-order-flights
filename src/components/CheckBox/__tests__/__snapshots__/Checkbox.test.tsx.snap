// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBox
    checked={false}
    label="Test"
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Test",
        <input
          checked={false}
          className="checkbox"
          onChange={[MockFunction]}
          type="checkbox"
        />,
        <span
          className="checkmark"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      "Test",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "className": "checkbox",
          "onChange": [MockFunction],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "checkmark",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Test",
          <input
            checked={false}
            className="checkbox"
            onChange={[MockFunction]}
            type="checkbox"
          />,
          <span
            className="checkmark"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        "Test",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "checkbox",
            "onChange": [MockFunction],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "checkmark",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<CheckBox /> should match snapshot 2`] = `"<label class=\\"container\\">Test<input type=\\"checkbox\\" class=\\"checkbox\\"/><span class=\\"checkmark\\"></span></label>"`;
